python/cudf/cudf/_libxx/quantiles.pyx

```diff
@@ -0,0 +1,51 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.
+
+# cython: profile=False
+# distutils: language = c++
+# cython: embedsignature = True
+# cython: language_level = 3
+
+import pandas as pd
+
+from cudf._libxx.lib cimport *
+from cudf._libxx.column cimport Column
+from cudf._libxx.table cimport Table
+cimport cudf._libxx.includes.quantiles as cpp_quantiles
+
+
+def quantiles(Table source_table, q, interp, is_input_sorted, column_order, null_precedence):
+    cdef unique_ptr[table] c_result
+    cdef table_view c_input = source_table.view()
+    cdef vector[double] c_q
+    cdef interpolation c_interp = <interpolation>(<interpolation_t> interp)
+    cdef sorted c_is_input_sorted = <sorted>(<sorted_t> is_input_sorted)
+    cdef vector[order] c_column_order
+    cdef vector[null_order] c_null_precedence
+
+    for value in column_order:
+        c_column_order.push_back(
+            <order>(<order_t> value)
+        )
+
+    for value in null_precedence:
+        c_null_precedence.push_back(
+            <null_order>(<null_order_t> value)
+        )
+
+    with nogil:
+        c_result = move(
+            cpp_quantiles.quantiles(
+                c_input,
+                c_q,
+                c_interp,
+                c_is_input_sorted,
+                c_column_order,
+                c_null_precedence
+            )
+        )
+
+    return Table.from_unique_ptr(
+        move(c_result),
+        column_names=source_table._column_names,
+        index_names=source_table._index._column_names
```

Comment:
I think we still need to guard against `source_table._index` being `None` here

Reply:
`copying.pyx: gather` does not take that approach. Update that as well?

Reply:
cc @shwina any insights here? Having to guard against `._index` being None makes this a bit of a pain. Should we have a `source_table._index_column_names` property or something that handles this for us?

Reply:
This might help:
https://github.com/rapidsai/cudf/blob/87a9e49d28c697f01c7a8ad7cefc5c700054373b/python/cudf/cudf/_libxx/table.pyx#L54-L56